File: web/app/models/Deal.js
"use strict";

function Deal(){
    this.id = '';
    this.name = '';
    this.address = '';
    this.category = '';
    this.prize = 0;
    this.distance = 0;
    this.openHours = {
      start: 0,
      end: 0
    };
    this.estimatedDeliver = 0;
    this.stock = 0;
    this.expiry = new Date();
};

File: src/java/lofi/models/Deal.java
package lofi.models;

import java.util.Date;

public class Deal {
    
    public class OpenHours{
        public int start;
        public int end;
    }
    
    public int id;
    public String productName;
    public String dealerName;
    public String dealerAddress;
    public double prize;
    public int distance;
    public OpenHours openHours;
    public int estimatedDeliver;
    public int stock;
    public Date expiry;
    
    public Deal(){
        openHours = new OpenHours();
    }
}
